### PR TITLE
WIP: download script for github releases

### PR DESCRIPTION
**Pull request by IgorTodorovskiIBM**
* Adds initial version of downloadports, which downloads the latest releases from Github and extracts to $PWD (by default) or a directory as specified by -d
* Also created an importenvs so that we can import envs from the ${HOME}zot/boot dir for dev use
---
* We may want to create a common library/script for functions like printSoftError, printError, etc?

### CODE REVIEW

File: bin/downloadports [hunk 1]
```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+#
+# Downloads and extracts the latest ported releases from GitHub
+
+printSoftError()
+{
+  printf "${NC}${RED}${BOLD}***ERROR: ${NC}${RED}${1}${NC}\n" >&2
```
Comment (MikeFultonDev) [inline bin/downloadports]: are these variables declared in this script? Do we need to pre-req bash?

File: bin/downloadports [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      ;;
+    *)
+      printError "Unknown option ${1} specified"
+      ;;
+  esac
+  shift;
+done
+
+if [ ! -z ${download} -a -d ${download} ]; then
+  pushd ${download}
```
Comment (MikeFultonDev) [inline bin/downloadports]: i think pushd/popd are specific to bash. Can we keep scripts general?

File: bin/downloadports [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+fi
+
+#FIXME: Once jq is ported, rewrite with jq
+for repo in $(curl -s "https://api.github.com/users/ZOSOpenTools/repos?per_page=100" | grep "\"full_name\":" | cut -d '"' -f 4); do
```
Comment (MikeFultonDev) [inline bin/downloadports]: do you want to catch the error if curl isn't able to download repos?

File: bin/downloadports [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  repo=${repo#"ZOSOpenTools/"}
+  echo $repo
+  if [ ! -z "${toolrepo}" -a "${toolrepo}" == "${repo}" ]; then
+    URL=$(curl -s https://api.github.com/repos/ZOSOpenTools/${repo}/releases/latest | grep browser_download_url | cut -d '"' -f 4)
```
Comment (MikeFultonDev) [inline bin/downloadports]: should check for error

File: bin/importenvs [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+#
+# Used for development to source all the boot .envs
+# . importenvs
+
+if [ ! -d "${HOME}/zot/boot/" ]; then
+  echo "boot directory does not exist.  Existing"
+  exit 4;
+fi
+
+for deps in $(find "${HOME}/zot/boot/" -name "*.env")
```
Comment (MikeFultonDev) [inline bin/importenvs]: don't you want to _just_ find .env (and not *.env ?)

File: bin/downloadports (hunk 1 above)
Comment (IgorTodorovskiIBM) [reply bin/downloadports]: added common.inc, which we can use as a library for our scripts

**Review by MikeFultonDev — APPROVED** [review]

**Pull request merged**